src/isar/mission_planner/task_selector_interface.py

```diff
@@ -0,0 +1,16 @@
+from abc import ABCMeta, abstractmethod
+from typing import List
+
+from isar.models.mission import Task
+
+
+class TaskSelectorInterface(metaclass=ABCMeta):
+    def __init__(self) -> None:
+        self.tasks: List[Task] = None
+
+    def initialize(self, tasks: List[Task]) -> None:
+        self.tasks = tasks
+
+    @abstractmethod
+    def next_task(self) -> Task:
```

Comment:
How do you indicate that all tasks are finished? Should it return None or raise an exception? I see that in the state machine we expect a StopIteration exception when there are no more tasks left. This works fine for the IterativeTaskSelector, but should probably be generalized with an exception that is not specific to iterators. Suggest adding some documentation on the expected behavior of the next_task method.

Reply:
Good point. Will generalize the stopping condition for updating the current task in the state machine.